skyrim-platform/src/platform_se/skyrim_platform/main.cpp

```diff
@@ -510,7 +519,9 @@ class SkyrimPlatformApp : public CEFUtils::SKSEPluginBase
 
     auto onProcessMessage = std::make_shared<ProcessMessageListenerImpl>();
 
-    overlayService = std::make_shared<OverlayService>(onProcessMessage);
+    std::vector<TextToDraw> textsToDraw = ObtainTextsToDraw();
+    overlayService =
+      std::make_shared<OverlayService>(onProcessMessage, &textsToDraw);
```

Comment:
Change constructor parameter to `std::function<std::vector<TextToDraw>>` and pass `ObtainTextsToDraw` itself instead of passing `textsToDraw`. It's because texts can change every frame.